src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,239 @@
+//#define TEST_GRPC
+//#define TEST_LCM
+#define TEST_API
+
+#include <bits/stdc++.h>
+#include <thread>
+
+#include <hyro/SignalGenerator.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/Thresholding.h>
+#include <hyro/DigitalConverterComponent.h>
+
+// We will work with state machines that wraps the components
+#include <hyro/core/StateMachine.h>
+#include <hyro/utils/components/PrintComponent.h>
+
+// A utility that will manage multi-threaded updates for us (less code to write)
+#include <hyro/utils/StateMachineSpinner.h>
+#include <hyro/utils/components/NumberGeneratorComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+#include <hyro/msgs/common/Basic.h>
+#include <hyro/widgets/WidgetCollection.h>
+
+using namespace hyro;
+using namespace std::string_literals;
+using namespace std::chrono_literals;
+
+double signal_property_amplitude;
```

Comment:
@silverwings88 please, avoid global variables.

Please, try always use local variables and arguments.